test/notifications/open_test.dart

```diff
@@ -178,6 +189,20 @@ void main() {
       }
     }
 
+    void takeHomePageReplacement(int accountId, int prevAccountId) {
+      check(poppedRoutes).last.which(
+        (it) => it.isA<MaterialAccountWidgetRoute>()
+          ..accountId.equals(prevAccountId)
+          ..page.isA<HomePage>());
+      // We only care about HomePage being the last popped route.
+      poppedRoutes.clear();
```

Comment:
This doesn't match the meaning of our other "takeFoo" helpers, in this or other tests: generally those all consume exactly the items that they check the details of, and consume them in the order they happened.

In this version, it not only ignores any non-HomePage routes before the last, but also any extra HomePage routes before the last one popped. So e.g. if a test popped two different HomePage routes and only then called this function, this would look only at the last one (but then in the next line it would look at the *first* route pushed, a potentially-confusing mismatch).

If you only ever want to look at the last route popped, probably clearest to only maintain that data: like `Route<void>? lastPoppedRoute`, instead of a list.